setup.yml

```diff
@@ -0,0 +1,174 @@
+---
+- name: Setup Hush Line application server
+  hosts: all
+  become: yes
+  vars_files:
+    - vars/common.yml
+    - vars/secrets/secrets.yml
+  tasks:
+    - name: Install required packages
+      apt:
+        name:
+          - python3
+          - python3-pip
+          - git
+          - nginx
+          - default-mysql-server
+          - python3-venv
+          - certbot
+          - python3-certbot-nginx
+          - libnginx-mod-http-geoip
+          - ufw
+          - fail2ban
+          - redis
+          - redis-server
+          - apt-transport-https
+          - curl
+          - sqlite3
+          - acl
+        update_cache: yes
+        state: latest
+
+    - name: Clone Hush Line repository
+      git:
+        repo: https://github.com/scidsg/hushline.git
+        dest: /var/www/html/hushline
+        version: ansible
+
+    - name: Install Poetry
+      shell: |
+        curl -sSL https://install.python-poetry.org | python3 -
+      args:
+        creates: /root/.local/bin/poetry
+
+    - name: Create virtual environment
+      shell: python3 -m venv /var/www/html/hushline/venv
+      args:
+        creates: /var/www/html/hushline/venv
+
+    - name: Configure Poetry to use existing virtual environment
+      shell: |
+        /root/.local/bin/poetry config virtualenvs.create false
+      args:
+        executable: /bin/bash
+
+    - name: Install dependencies with Poetry
+      shell: |
+        source /var/www/html/hushline/venv/bin/activate && /root/.local/bin/poetry install
+      args:
+        chdir: /var/www/html/hushline
+        executable: /bin/bash
+
+    - name: Perform database migrations
+      shell: |
+        source /var/www/html/hushline/venv/bin/activate && flask db upgrade
+      args:
+        chdir: /var/www/html/hushline
+        executable: /bin/bash
+      become: no
+      become_user: hushlineuser
+
+    - name: Create Hushline user
+      user:
+        name: hushlineuser
+        system: yes
+        shell: /usr/sbin/nologin
+        group: www-data
+        create_home: no
+
+    - name: Ensure SQLite directory exists
+      file:
+        path: /var/lib/hushline
+        state: directory
+        owner: hushlineuser
+        group: www-data
+
+    - name: Ensure /etc/hushline directory exists
+      file:
+        path: /etc/hushline
+        state: directory
+        owner: hushlineuser
+        group: www-data
+        mode: '0755'
```

Comment:
Mode `755` is unsafe as this directory will have sensitive configurations. Should be `0700`